src/scripts/modal.js

```diff
@@ -0,0 +1,23 @@
+function modal() {
+  const heroContent = document.querySelector(".hero__content");
+  const modalContent = document.querySelector("#modal");
+  const checkOfferBtn = document.querySelector(".hero--cta");
+  const confrimModalBtn = document.querySelector(".modal-btn--confirm");
+  const closeModalBtn = document.querySelector(".modal-btn--close");
+  const closeModalIcon = document.querySelector(".modal-icon--close");
```

Comment:
No need to traverse whole document looking for those elements. Beside that `querySelector` returns first matching element, so it can easily return different element than expected. 

This way is much safer:
```ts
const confirmBtn = modalContent.querySelector("...");
```

Reply:
Changed.